src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -222,7 +223,7 @@ def post_fork_child(self):
 
     # Broadcast our process group ID (in PID form - i.e. negated) to the remote client so
     # they can send signals (e.g. SIGINT) to all processes in the runners process group.
-    NailgunProtocol.send_pid(self._socket, bytes(os.getpgrp() * -1))
+    NailgunProtocol.send_pid(self._socket, bytes(str(os.getpgrp() * -1), 'ascii'))
```

Comment:
@kwlzn and I discussed this in the Python3 slack channel. We're first calling `str()` to cast the int to a str, then `bytes()` because Nailgun is expecting bytes with its IO